src/index.ts

```diff
@@ -20,22 +20,68 @@ import mapperReader from './mapperReader.js';
 import { MapperServer } from './mapperServer.js';
 import { BidiServerRunner } from './bidiServerRunner.js';
 import { IServer } from './iServer.js';
+import argparse from 'argparse';
+
+function parseArguments() {
+  var parser = new argparse.ArgumentParser({
+    add_help: true,
+    exit_on_error: false,
+  });
+
+  parser.add_argument('-p', '--port', {
+    help: 'Port, BiDi server should listen to. Default is 8080.',
+    default: 8080,
+  });
+
+  parser.add_argument('-hl', '--headless', {
+    help: 'Sets if browser should run in headless or headful mode. Default is `--headless=true`.',
+    default: true,
+  });
+
+  parser.add_argument('-b', '--browser', {
+    help: 'Path to browser executable.',
+    required: true,
+  });
+
+  // `parse_known_args` puts known args in the first element of the result.
+  const args = parser.parse_known_args();
+  return args[0];
+}
 
 (async () => {
   try {
     console.log('Launching BiDi server.');
-    BidiServerRunner.run(_onNewBidiConnectionOpen, _onBidiConnectionClosed);
+
+    const args = parseArguments();
+    const _bidiPort = parseInt(args.port);
```

Comment:
Hopefully the `parseInt` can be avoided here by pushing the type into the argument definition.

Also, leading underscores for local variables seems odd, they're always "private" in some sense and can't be accidentally used elsewhere, which I guess is why `_onNewBidiConnectionOpen` is prefixed.

Reply:
done